openapps/opencoap/opencoap.c

```diff
@@ -849,19 +867,36 @@ owerror_t opencoap_options_encode(
     return E_SUCCESS;
 }
 
-coap_option_iht* opencoap_find_option(coap_option_iht* array, uint8_t arrayLen, coap_option_t option) {
-    uint8_t i;
 
-    if (array == NULL || arrayLen == 0) {
-        return NULL;
-    }
 
-    for (i = 0; i < arrayLen; i++) {
-        if (array[i].type == option) {
-            return &array[i];
+uint8_t opencoap_find_option(coap_option_iht* array, uint8_t arrayLen, coap_option_t option, uint8_t* startIndex) {
+   uint8_t i;
+   uint8_t j = 0;
```

Comment:
A small detail, could you initialize `j` and `found` on a separate line. That's a convention that we generally use in OpenWSN.